jni/lib/src/nio/jbyte_buffer.dart

```diff
@@ -0,0 +1,201 @@
+        JniCallType.objectType,
+        [array.reference, JValueInt(offset), JValueInt(length)],
+      ).object,
+    );
+  }
+
+  static final _sliceId = Jni.accessors
+      .getMethodIDOf(_class.reference, r"slice", r"()Ljava/nio/ByteBuffer;");
+
+  /// Creates a new byte buffer whose content is a shared subsequence of this
+  /// buffer's content.
+  JByteBuffer slice() {
+    return const JByteBufferType().fromRef(Jni.accessors.callMethodWithArgs(
+        reference, _sliceId, JniCallType.objectType, []).object);
+  }
+
+  static final _duplicateId = Jni.accessors.getMethodIDOf(
+      _class.reference, r"duplicate", r"()Ljava/nio/ByteBuffer;");
+
+  /// Creates a new byte buffer that shares this buffer's content.
+  JByteBuffer duplicate() {
+    return const JByteBufferType().fromRef(Jni.accessors.callMethodWithArgs(
+        reference, _duplicateId, JniCallType.objectType, []).object);
+  }
+
+  static final _asReadOnlyBufferId = Jni.accessors.getMethodIDOf(
+      _class.reference, r"asReadOnlyBuffer", r"()Ljava/nio/ByteBuffer;");
+
+  /// Creates a new, read-only byte buffer that shares this buffer's content.
+  JByteBuffer asReadOnlyBuffer() {
+    return const JByteBufferType().fromRef(Jni.accessors.callMethodWithArgs(
+        reference, _asReadOnlyBufferId, JniCallType.objectType, []).object);
+  }
+
+  static final _getId =
+      Jni.accessors.getMethodIDOf(_class.reference, r"get", r"()B");
+
+  /// Reads the byte at this buffer's current [position], and then increments the
+  /// [position].
+  ///
+  /// Throws:
+  ///  * [BufferOverflowException] - If the buffer's current [position] is not
+  ///    smaller than its [limit]
+  int get nextByte {
+    return Jni.accessors
+        .callMethodWithArgs(reference, _getId, JniCallType.byteType, []).byte;
+  }
+
+  static final _putId = Jni.accessors
+      .getMethodIDOf(_class.reference, r"put", r"(B)Ljava/nio/ByteBuffer;");
+
+  /// Writes the given byte into this buffer at the current [position], and then
+  /// increments the [position].
+  ///
+  /// Throws:
+  /// * [BufferOverflowException] - If this buffer's current [position] is not
+  ///   smaller than its [limit]
+  /// * [ReadOnlyBufferException] - If this buffer is read-only
+  set nextByte(int b) {
+    Jni.env.DeleteGlobalRef(Jni.accessors.callMethodWithArgs(
+        reference, _putId, JniCallType.objectType, [JValueByte(b)]).object);
+  }
+
+  static final _arrayId =
+      Jni.accessors.getMethodIDOf(_class.reference, r"array", r"()[B");
+
+  @override
+  JArray<jbyte> get array {
+    return const JArrayType(jbyteType()).fromRef(Jni.accessors
+        .callMethodWithArgs(
+            reference, _arrayId, JniCallType.objectType, []).object);
+  }
+
+  /// Returns this byte buffer as a [Uint8List].
+  ///
+  /// Throws [ArgumentError] if the buffer is not direct or the JVM does not
+  /// support the direct buffer operations.
+  Uint8List asUint8List() {
+    assert(isDirect, 'The buffer must be direct.');
```

Comment:
if (!isDirect) {
  throw StateError('The buffer must be created with JByteBuffer.allocateDirect');
}